testutil/deepequal.go

```diff
@@ -0,0 +1,129 @@
+package testutil
+
+import "reflect"
+
+// DeepEqual is similar to reflect.DeepEqual, but treats
+// nil and []T{} as equal.
+// (Note, it's also a more naive implementation that doesn't detect
+// cycles).
+func DeepEqual(x, y interface{}) bool {
+	vx := reflect.ValueOf(x)
+	vy := reflect.ValueOf(y)
+	return deepValueEqual(vx, vy)
+}
+
+func deepValueEqual(x, y reflect.Value) bool {
+	if isNilish(x) && isNilish(y) {
```

Comment:
No alas; that breaks the all-important case of `nil` (which is not "valid") vs. `[]byte{}` (which is).